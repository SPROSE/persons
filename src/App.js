/**
 * author: richard sproson
 * description: App renders basic containers > hits endpoints > stores data > passes to props
 */

/* Modules */
import React, { useState, useEffect } from 'react';

/* Styles */
import './App.scss';

/* Components */
import DisplayPersons from './components/display-persons';

function App() {
    const [hasError, setErrors] = useState('');
    const [personsData, setPersonsData] = useState([]);
    const [personsDetailsData, setPersonDetailsData] = useState([]);

    const fetchData = async(url, action) => {
        const data = await fetch(url);
        data
            .json()
            .then(data => action(data.value))
            .catch(err => setErrors(err));
    };

    useEffect(() => {
        fetchData('https://services.odata.org/Experimental/OData/OData.svc/Persons', setPersonsData);
        fetchData('https://services.odata.org/Experimental/OData/OData.svc/PersonDetails', setPersonDetailsData);
    }, []);

    if (hasError.length) {
        return (
            <div>There was an error fetching data {hasError}</div>
        )
    }
    return (
        <div className="App">
            <header className="App-header">
                <h1>Persons API</h1>
            </header>
            <section>
                <DisplayPersons
                    personsData={personsData}
                    personsDetailsData={personsDetailsData}
                />
            </section>
        </div>
    );
}

export default App;
