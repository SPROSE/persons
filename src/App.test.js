import React from 'react';
import { render } from '@testing-library/react';
import { shallow } from 'enzyme';

import App from './App';


test('renders header elements', () => {
    const { getByText } = render(<App />);
    const headerText = getByText('Persons API');
    expect(headerText).toBeInTheDocument();
});

describe('<App /> without props', () => {
    const container = shallow(<App />);
    it('should match the snapshot', () => {
        expect(container.html()).toMatchSnapshot();
    });
});