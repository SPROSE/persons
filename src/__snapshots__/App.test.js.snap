// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> without props should match the snapshot 1`] = `"<div class=\\"App\\"><header class=\\"App-header\\"><h1>Persons API</h1></header><section><h2>Persons Data</h2><div>Loading data...</div></section></div>"`;
